### PR TITLE
feat: add hugo preset

### PR DESCRIPTION
**Pull request by tiagokrebs**
Adding a new Hugo preset.  

About not using the `hugo` binary directly but through the package manager.
When you think about a typical Hugo site, the easiest example of dependencies are themes. But there’s something more fundamental to realize: Hugo itself is a dependency. The Hugo binary is a 3rd party library that is required for your application to work. If we could install Hugo locally as a dependency, we would have full control over the Hugo version used in our project. And best of all, all our project users would use the same version.  

The solution is to use a binary wrapper for Hugo with a package.json file already added. [hugo-bin](https://github.com/fenneclab/hugo-bin) is currently the most popular/accepted package.

### CODE REVIEW

File: lib/presets/custom/hugo/deliver/prebuild.js [hunk 1]
```diff
@@ -0,0 +1,21 @@
...
+async function prebuild() {
+  const newOutDir = '.edge/storage';
+  let outDir = 'public';
+
+  const command = packageManager === 'npm' ? 'npx' : packageManager;
+  await exec(`${command} hugo`, 'Hugo', true);
+
+  // move files to vulcan default path
+  copyDirectory(outDir, newOutDir);
+  rm(outDir, { recursive: true, force: true });
```
Comment (MagnunAVFAzion) [inline lib/presets/custom/hugo/deliver/prebuild.js:18]: We need to generate the manifest at the end of the prebuild step. You can copy this from https://github.com/aziontech/vulcan/blob/stage/lib/presets/custom/react/deliver/prebuild.js#L14 (lines 14-20) as this preset is purely static (deliver mode).

**Review by MagnunAVFAzion — CHANGES_REQUESTED** [review]

File: lib/presets/custom/hugo/deliver/prebuild.js [hunk 2]
```diff
@@ -0,0 +1,21 @@
...
+
+/**
+ * Runs custom prebuild actions
+ */
+async function prebuild() {
+  const newOutDir = '.edge/storage';
+  let outDir = 'public';
+
+  const command = packageManager === 'npm' ? 'npx' : packageManager;
+  await exec(`${command} hugo`, 'Hexo', true);
```
Comment (jotanarciso) [inline lib/presets/custom/hugo/deliver/prebuild.js]: 'Hexo' -> 'Hugo'

**Comment by tiagokrebs** [comment]
@jotanarciso @MagnunAVFAzion Now we have one example of Hugo here too.

**Review by jotanarciso — APPROVED** [review]

**Comment by jotanarciso** [comment]
Nice!

**Review by MagnunAVFAzion — APPROVED** [review]

**Pull request merged**